rocAL/rocAL/include/pipeline/node.h

```diff
@@ -41,8 +41,9 @@ class Node
     void add_next(const std::shared_ptr<Node>& node) {} // To be implemented
     void add_previous(const std::shared_ptr<Node>& node) {} //To be implemented
     std::shared_ptr<Graph> graph() { return _graph; }
-    void set_meta_data(pMetaDataBatch meta_data_info){ _meta_data_info = meta_data_info;}
+    void set_meta_data(pMetaDataBatch meta_data_info) { _meta_data_info = meta_data_info; }
     bool _is_ssd = false;
+    void create_crop_tensor(vx_tensor &crop_tensor, void **crop_coordinates);
```

Comment:
why are we creating a crop_tensor in node.h. This will be created for all the nodes which are derived

Reply:
- The crop_tensor is not derived for all nodes. As they are created and maintained only by the derived crop nodes.
- The crop nodes handles the allocation and release.
- Only the function to create a crop tensor is present in node.h since we wanted to keep it common to all crop nodes we have introduced the API and it's definition here.
- This function accepts the vx_tensor and the ptr by reference which is created in the crop nodes which is derived from base node class and allocates the tensor passed by the derived nodes.